app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/InterestGroupHandle.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * mailchimp-lib Magento Component
+ *
+ * @category  Ebizmarts
+ * @package   mailchimp-lib
+ * @author    Ebizmarts Team <info@ebizmarts.com>
+ * @copyright Ebizmarts (http://ebizmarts.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+class Ebizmarts_MailChimp_Model_Api_Subscribers_InterestGroupHandle
+{
+    /**
+     * @var Ebizmarts_MailChimp_Helper_Data
+     */
+    protected $_helper;
+    /**
+     * @var Ebizmarts_MailChimp_Helper_Date
+     */
+    protected $_dateHelper;
+
+    public function __construct()
+    {
+        $this->_helper = Mage::helper('mailchimp');
+        $this->_dateHelper = Mage::helper('mailchimp/date');
+    }
+
+    /**
+     * @param $groupings
+     * @param $customerData
+     * @param $listId
+     * @param bool $isDataSubscriber
+     * @throws Ebizmarts_MailChimp_Helper_Data_ApiKeyException
+     * @throws MailChimp_Error
+     * @throws MailChimp_HttpError
+     */
+    public function processGroupsData($groupings, $customerData, $listId, $isDataSubscriber = false)
```

Comment:
Please make all the parameters class attributes.
$isDataSubscribers could be removed making a more elaborated use of $customerData object.
$customerData can be divided in two attributes: $subscriber and $customer initializing them in null. Then you can do **if ($subscriber !== null)** to know which way you can get the customer Id.
All this can be handled in a getCustomerId function to make it easier for understanding. The same for subscriberId.